src/main/java/delete_post/use_case/DeletePostInteractor.java

```diff
@@ -0,0 +1,37 @@
+package delete_post.use_case;
+
+import delete_post.interface_adapters.DeletePostPresenter;
+import entities.User;
+
+public class DeletePostInteractor implements DeletePostInputBoundary{
+
+    private final DeletePostPresenter presenter;
+    private final DeletePostDsGateway dataAccess;
+
+    public DeletePostInteractor(DeletePostPresenter presenter, DeletePostDsGateway dataAccess){
+        this.presenter = presenter;
+        this.dataAccess = dataAccess;
+    }
+
+    @Override
+    public void delete(DeletePostRequestModel requestModel){
+        if (requestModel.getIsTimer() || requestModel.getIsAdmin() ||
```

Comment:
Important - the delete post interactor cannot call the presenter if it has been called by the timer. That way, we can have the two cases completely separate while using the same functionality